ThunkLibs/libVDSO/libVDSO_Guest.cpp

```diff
@@ -0,0 +1,39 @@
+
+/*
+$info$
+tags: thunklibs|VDSO
+desc: Linux VDSO thunking
+$end_info$
+*/
+
+#include <stdio.h>
+#include <cstring>
+
+#include <sched.h>
+#include <sys/time.h>
+#include <time.h>
+
+#include "common/Guest.h"
+
+// Hardcoded VDSO
+MAKE_THUNK(libVDSO, time, "0x52, 0xcd, 0xab, 0xa1, 0xf7, 0xb0, 0xdd, 0xff, 0xd3, 0x68, 0x09, 0xb0, 0xce, 0xdb, 0xe9, 0xfb, 0x3a, 0xf6, 0xfc, 0x8d, 0xff, 0x3f, 0xf4, 0xb2, 0x0c, 0xcc, 0x1f, 0xe3, 0x99, 0x14, 0x77, 0xcf")
+MAKE_THUNK(libVDSO, gettimeofday, "0x99, 0x93, 0xf1, 0xff, 0x97, 0x08, 0xf4, 0xfe, 0x49, 0xd7, 0x73, 0x9c, 0x43, 0x6b, 0x9e, 0x89, 0x84, 0x53, 0xd3, 0xde, 0x32, 0x11, 0x98, 0x28, 0xb5, 0xbb, 0x95, 0xcb, 0xb4, 0x72, 0xf4, 0x18")
+MAKE_THUNK(libVDSO, clock_gettime, "0x42, 0xb4, 0xf8, 0xd8, 0xf3, 0xc2, 0xb2, 0x8f, 0x6d, 0x40, 0x35, 0x59, 0x1f, 0x53, 0x9c, 0x2d, 0xa0, 0xfa, 0x1c, 0x3a, 0x60, 0x91, 0xe7, 0x84, 0x0c, 0x50, 0x93, 0x31, 0xd9, 0x4f, 0xb6, 0xff")
+MAKE_THUNK(libVDSO, clock_getres, "0xeb, 0x82, 0x86, 0x04, 0x41, 0x72, 0xbd, 0x2c, 0xfb, 0x83, 0xa7, 0x64, 0x5a, 0xca, 0x49, 0x7e, 0x00, 0xf5, 0xb3, 0x73, 0xfa, 0xd4, 0xf1, 0x67, 0x8b, 0x54, 0x49, 0x7d, 0xb3, 0x33, 0x88, 0x1b")
+MAKE_THUNK(libVDSO, getcpu, "0x1f, 0xaa, 0x53, 0xd3, 0x21, 0x98, 0x63, 0xb4, 0xcc, 0xbd, 0x61, 0x29, 0x72, 0xb2, 0x64, 0xd2, 0xe6, 0xe3, 0x9d, 0x0a, 0x4c, 0x4f, 0x9e, 0x70, 0x0b, 0x29, 0x76, 0xb1, 0xf0, 0x21, 0xf0, 0x43")
+
+#include "function_packs.inl"
+
+extern "C" {
+time_t __vdso_time(time_t *tloc) __attribute__((alias("fexfn_pack_time")));
+int __vdso_gettimeofday(struct timeval *tv, struct timezone *tz) __attribute__((alias("fexfn_pack_gettimeofday")));
+int __vdso_clock_gettime(clockid_t, struct timespec *) __attribute__((alias("fexfn_pack_clock_gettime")));
+int __vdso_clock_getres(clockid_t, struct timespec *) __attribute__((alias("fexfn_pack_clock_getres")));
+int __vdso_getcpu(uint32_t *, uint32_t *) __attribute__((alias("fexfn_pack_getcpu")));
+
+__attribute__((alias("fexfn_pack_time"))) auto time(time_t * a_0) -> time_t;
```

Comment:
Why can't these use the generated `function_packs_public.inl` instead?

Reply:
You're right, resolved.